cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SubqueryExpressionsTest.scala

```diff
@@ -140,5 +144,107 @@ class SubqueryExpressionsTest extends DocumentingTest {
         }
       }
     }
+
+    section("COUNT subqueries", "count-subqueries") {
+      p("An count subquery can be used to find out how many times a specified pattern appears in the data.")
+      functionWithCypherStyleFormatting(
+        """COUNT {
+          #  [Pattern]
+          #  WHERE [Expression]
+          #}""".stripMargin('#'))
+      p("It is worth noting that the `MATCH` keyword can be omitted in subqueries and that the `WHERE` clause is optional.")
+
+      section("Simple count subquery", "count-subquery-simple-case") {
+        p(
+          """Variables introduced by the outside scope can be used in the `COUNT` subquery without importing them
+            #<<subquery-correlated-importing|as necessary with `CALL` subqueries>>.
+            #The following query exemplifies this and outputs the owners of more than one dog:
+            #""".stripMargin('#'))
+        query(
+          """MATCH (person:Person)
+            #WHERE COUNT { (person)-[:HAS_DOG]->(:Dog) } > 1
+            #RETURN person.name AS name""".stripMargin('#'),
+          ResultAssertions(r => {
+            r.toList should equal(List(Map("name" -> "Peter")))
+          })) {
+          resultTable()
+        }
+      }
+      section("Count subquery with `WHERE` clause", "count-subquery-with-where") {
+        p(
+          """A `WHERE` clause can be used inside the `COUNT` pattern.
+            #Variables introduced by the `MATCH` clause and the outside scope can be used in this scope.""".stripMargin('#'))
+        query(
+          """MATCH (person:Person)
+            #WHERE COUNT {
+            #  (person)-[:HAS_DOG]->(dog:Dog)
+            #  WHERE person.name = dog.name
+            #} = 1
+            #RETURN person.name AS name""".stripMargin('#'),
+          ResultAssertions(r => {
+            r.toList should equal(List(Map("name" -> "Andy")))
+          })) {
+          resultTable()
+        }
+      }
+      section("Using count subqueries inside other clauses", "count-subqueries-other-clauses") {
+        p("COUNT can be used in any position in a query, with the exception of administration commands, where it is restricted. We provide a few examples below.")
+        section("Using count in RETURN", "count-subqueries-with-return") {
+          query(
+            """MATCH (person:Person)
+              #RETURN person.name, COUNT { (person)-[:HAS_DOG]->(:Dog) } as howManyDogs
+      """.stripMargin('#'),
+            ResultAssertions(r => {
+              r.toList should equal(List(Map("howManyDogs" -> 1, "person.name" -> "Andy"), Map("howManyDogs" -> 0, "person.name" -> "Timothy"), Map("howManyDogs" -> 2, "person.name" -> "Peter")))
+            })) {
+            resultTable()
+          }
+        }
+        section("Using count in SET", "count-subqueries-with-set") {
+          query(
+            """MATCH (person:Person) WHERE person.name ="Andy"
+              #SET person.howManyDogs = COUNT { (person)-[:HAS_DOG]->(:Dog) } + 1
```

Comment:
This `+ 1` is probably to indicate that one can also use this in algebraic expressions but it might suggest to some readers that `COUNT` would result in a too low value.

Reply:
I'll remove the `+1`, I think that is less confusing.

Reply:
:+1: